allauth/headless/mfa/views.py

```diff
@@ -50,8 +48,14 @@ class ManageTOTPView(AuthenticatedAPIView):
     def get(self, request, *args, **kwargs):
         authenticator = self._get_authenticator()
         if not authenticator:
+            adapter = get_adapter()
             secret = totp.get_totp_secret(regenerate=True)
-            return response.TOTPNotFoundResponse(request, secret)
+            totp_url: str = totp.build_totp_url(
+                adapter.get_totp_label(request.user),
+                adapter.get_totp_issuer(),
+                secret,
+            )
```

Comment:
Building this TOTP URI follows the same pattern as in the [headed view](https://github.com/mecampbellsoup/django-allauth/blob/4ee8a1157ea55afd932441aff89ea2dff1821803/allauth/mfa/views.py#L112-L116).

Reply:
How about we add `def build_totp_url(self, secret):` to the adapter method, then we don't have this duplication, and also allow for people to intervene and alter the URL as they see fit?

Reply:
Done in https://github.com/pennersr/django-allauth/commit/18c7ca4701f8891e36808dae47bbba75de92af37?

Reply:
The comment is about `build_topt_url` -- that is not in 18c7ca4.